pkg/placement/ha_test.go

```diff
@@ -94,17 +101,27 @@ func TestPlacementHA(t *testing.T) {
 	})
 
 	t.Run("retrieve state in follower", func(t *testing.T) {
-		follower := (findLeader(t, raftServers) + 1) % 3
+		var follower, oldLeader int
+		oldLeader = findLeader(t, raftServers)
+		follower = (oldLeader + 1) % 3
 		retrieveValidState(t, raftServers[follower], testMembers[0])
-	})
 
-	t.Run("new leader after leader fails", func(t *testing.T) {
-		oldLeader := findLeader(t, raftServers)
-		raftServerCancel[oldLeader]()
-		raftServers[oldLeader] = nil
-		require.Eventually(t, func() bool {
-			return oldLeader != findLeader(t, raftServers)
-		}, time.Second*10, time.Millisecond*100)
+		t.Run("new leader after leader fails", func(t *testing.T) {
+			raftServerCancel[oldLeader]()
+			raftServers[oldLeader] = nil
+
+			// It is painful that we have to include a `time.Sleep` here, but due to
+			// the non-deterministic behaviour of the raft library we are using we will
+			// later fail on slower test runner machines. A clock timer wait means we
+			// have a _better_ chance of being in the right spot in the state machine
+			// and the network has died down. Ideally we should move to a different
+			// raft library that is more deterministic and reliable for our use case.
+			time.Sleep(time.Second * 3)
+
+			require.Eventually(t, func() bool {
+				return oldLeader != findLeader(t, raftServers)
+			}, time.Second*10, time.Millisecond*100)
+		})
```

Comment:
for any `findLeader` calls, is `require.Eventually` needed?
Since `findLeader` itself has that for 10s with step being 1s. 
Here the wrapping `require.Eventually` is 10s with step as 100ms. Is this needed?

Reply:
It is needed in the case when we are waiting for the leader to change.